specifyweb/frontend/js_src/lib/querytask.js

```diff
@@ -289,7 +289,8 @@ const commonText = require('./localization/common').default;
                     .filter(function(f) { return f.spqueryfield.get('isdisplay'); })
                     .sortBy(function(f) { return f.spqueryfield.get('position'); })
                     .pluck('fieldSpec')
-                    .value()
+                    .value(),
+                linkField: this.query.get('selectdistinct') ? -1 : 0,
```

Comment:
I think it is confusing to use `-1` as a sentinel value here. I would prefer the type to be Optional[Int] and using `null` to indicate no value.